packages/plugins/opentelemetry/src/setup.ts

```diff
@@ -0,0 +1,67 @@
+// This must be run first. Node uses patching on implementations to inject telemetry so must run before the packages
+// are instantiated.
+import * as api from '@opentelemetry/api';
+import { propagation } from '@opentelemetry/api';
+import { AsyncLocalStorageContextManager } from '@opentelemetry/context-async-hooks';
+import { CompositePropagator, W3CTraceContextPropagator } from '@opentelemetry/core';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { DnsInstrumentation } from '@opentelemetry/instrumentation-dns';
+import { GrpcInstrumentation } from '@opentelemetry/instrumentation-grpc';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { BatchSpanProcessor } from '@opentelemetry/sdk-trace-node';
+import { SDK_INFO } from '@opentelemetry/core';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import {Resource} from "@opentelemetry/resources";
+import {
+    ATTR_SERVICE_NAME,
+    ATTR_TELEMETRY_SDK_LANGUAGE, ATTR_TELEMETRY_SDK_VERSION
+} from "@opentelemetry/semantic-conventions";
+import {OTLPTraceExporter} from "@opentelemetry/exporter-trace-otlp-grpc";
+
+const contextManager = new AsyncLocalStorageContextManager().enable();
+api.context.setGlobalContextManager(contextManager);
+
+const exporter = new OTLPTraceExporter({
+    url: "todo:configuration",
```

Comment:
We would need to drive this from configuration. I have done this within our gateway implementation but that has a different approach to the hive-gateway to config so Im not sure of the best approach.